### PR TITLE
Attach teacher to jornal

### PR DESCRIPTION
**Pull request by MalynaMarian**

### CODE REVIEW

File: src/app/admin-panel/admin-panel/attach-teacher/attach-teacher.component.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  filteredClasses: Group[];
+
+  object: AttachedTeacher = {
+    classId: 0,
+    subjectId: 0,
+    teacherId: 0
+  };
+  constructor(private attachService: HttpAttachTeacherService) { }
+
+  filterTeachers(event) {
```
Comment (dmitryy90) [inline src/app/admin-panel/admin-panel/attach-teacher/attach-teacher.component.ts]: add comments to methods

Comment (dmitryy90) [inline src/app/admin-panel/admin-panel/attach-teacher/attach-teacher.component.ts]: use typing

File: src/app/admin-panel/admin-panel/attach-teacher/attach-teacher.component.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  filterSubjects(event) {
+    this.filteredSubjects = [];
+    for (let i = 0; i < this.subjects.length; i++) {
+      const subject = this.subjects[i];
+      if (subject.subjectName.toLowerCase().indexOf(event.query.toLowerCase()) === 0) {
+        this.filteredSubjects.push(subject);
+      }
+    }
+  }
+  filterClasses(event) {
```
Comment (dmitryy90) [inline src/app/admin-panel/admin-panel/attach-teacher/attach-teacher.component.ts]: format file

File: src/app/admin-panel/admin-panel/attach-teacher/attach-teacher.component.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    }
+  }
+  postAttachTeacher() {
+    this.object.subjectId = this.form.value.subject.subjectId;
+    this.object.teacherId = this.form.value.teacher.id;
+    this.object.classId = this.form.value._class.id;
+    console.log(this.object);
+    this.attachService.postAttachTeacher(this.object)
+      .subscribe(
+        data => console.log(data),
```
Comment (dmitryy90) [inline src/app/admin-panel/admin-panel/attach-teacher/attach-teacher.component.ts]: if something is going to be here add TODO comment

File: src/app/models/attached-teacher.model.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export class AttachedTeacher {
+    classId: number;
```
Comment (dmitryy90) [inline src/app/models/attached-teacher.model.ts]: indentation

File: src/app/models/teacher.model.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+export class Teacher {
+    avatar: string;
```
Comment (dmitryy90) [inline src/app/models/teacher.model.ts]: indentation

File: src/app/services/http-attach-teacher.service.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  providedIn: 'root'
+})
+export class HttpAttachTeacherService {
+
+  private urlTeachers = 'https://fierce-shore-32592.herokuapp.com/teachers';
+  private urlSubjects = 'https://fierce-shore-32592.herokuapp.com/subjects';
+  private urlClases = 'https://fierce-shore-32592.herokuapp.com/classes';
+
+  constructor(private http: HttpClient) { }
+  // I have some questions
```
Comment (dmitryy90) [inline src/app/services/http-attach-teacher.service.ts]: add TODO comment

File: src/app/services/http-attach-teacher.service.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  //     tap(_ => {console.log(_);
+  //      const newArray = _.map(function (teacher) {
+  //        teacher.fullname = `${teacher.lastname} ${teacher.firstname}`;
+  //           return teacher; });
+  //           console.log(newArray);
+  //      return newArray;
+  //     })
+  //   );
+  // }
+  getTeachers(): Observable<Teacher[]> {
```
Comment (dmitryy90) [inline src/app/services/http-attach-teacher.service.ts]: format file

File: src/app/admin-panel/admin-panel/attach-teacher/attach-teacher.component.ts (hunk 2 above)
Comment (MalynaMarian) [reply src/app/admin-panel/admin-panel/attach-teacher/attach-teacher.component.ts]: use any

File: src/app/admin-panel/admin-panel/attach-teacher/attach-teacher.component.ts (hunk 1 above)
Comment (MalynaMarian) [reply src/app/admin-panel/admin-panel/attach-teacher/attach-teacher.component.ts]: use any

**Pull request merged**